### PR TITLE
Support adding quotes in rooms

### PR DESCRIPTION
**Pull request by mia-pi-git**
This is a thing a lot of people liked with Kid A, and since it's gone i figured it would be nice to migrate and not too much work.

### CODE REVIEW

File: server/chat-commands/info.ts
Comment (KrisXV) [inline server/chat-commands/info.ts:2773]:
```suggestion
		let buffer = `<div class="pad">`;
		if (!room.settings.quotes?.length) {
			return `${buffer}<h2>This room has no quotes.</h2></div>`;
		}

		let buffer = `<h2>Quotes for ${room.title}: (${room.settings.quotes.length})</h2>`;
```

**Review by AnnikaCodes — COMMENTED** [review]
I sent you a review on Discord as requested.

**Comment by Zarel** [comment]
Hm, I'm starting to worry if we're stuffing too much stuff in `room.settings`. I suppose at least we might want to consider having one file per room rather than one `chatrooms.json` for all the rooms.

That'll be dealt with later, though; I believe this is ready to merge. @AnnikaCodes should I be concerned about your private review?

**Review by Zarel — APPROVED** [review]

File: server/chat-commands/info.ts
Comment (Zarel) [inline server/chat-commands/info.ts:2537]: Wait, this is wrong. `Utils.shuffle` is a mutating shuffle which should not be applied to the quotes list. If there's no `Utils.random` you should be doing:

```suggestion
		const quote = room.settings.quotes[Math.floor(Math.random() * room.settings.quotes.length)];
```

File: server/chat-commands/info.ts
Comment (Zarel) [inline server/chat-commands/info.ts:2557]: This should probably at least track the user who added it, and the date it was added.

**Comment by AnnikaCodes** [comment]
> Hm, I'm starting to worry if we're stuffing too much stuff in `room.settings`. I suppose at least we might want to consider having one file per room rather than one `chatrooms.json` for all the rooms.
> 
> That'll be dealt with later, though; I believe this is ready to merge. @AnnikaCodes should I be concerned about your private review?

can you wait to merge this until I approve (or I can merge it then if you're ok with that)? I think it's probably good to go but I want to make sure that nothing from the private review slipped by and I need to go do stuff IRL for a bit.

File: server/chat-commands/info.ts
Comment (Zarel) [inline server/chat-commands/info.ts:2556]: Pretty low maximum... maybe this should be stored in SQLite?

**Comment by Zarel** [comment]
Nah, don't merge without me. I'll wait until I get an approval review from you before merging, though.

**Comment by KrisXV** [comment]
> Hm, I'm starting to worry if we're stuffing too much stuff in `room.settings`. I suppose at least we might want to consider having one file per room rather than one `chatrooms.json` for all the rooms.
> 
> That'll be dealt with later, though; I believe this is ready to merge. @AnnikaCodes should I be concerned about your private review?

Though not entirely related to the PR at hand, I'm unsure why we are using `room.settings` to store things that are not settings. I think we should either find a new place to store quotes/repeats/etc. or rename `settings` to something that makes more sense.

**Comment by AnnikaCodes** [comment]
Maybe have `room.settings` continue storing settings, and add a `room.data` attribute that stores data (with its own saving method, which could save to a separate file for each chatroom if desired). Probably we would also only support data for non-personal rooms.

**Review by AnnikaCodes — APPROVED** [review]

File: server/chat-commands/info.ts
Comment (Zarel) [inline server/chat-commands/info.ts:2541]: Do most quote databases show this data?

Comment (mia-pi-git) [reply server/chat-commands/info.ts:2541]: No, but i though that's what you meant by attribution? 
Since if you're not showing it here, there's little point, considering modlog exists.

Comment (Zarel) [reply server/chat-commands/info.ts:2541]: I don't think it needs to show author and date by default.

Comment (mia-pi-git) [reply server/chat-commands/info.ts:2541]: Okay.

Comment (mia-pi-git) [reply server/chat-commands/info.ts]: Updated. It will now only show who it's by / when if you do !randquote showauthor.

Comment (Zarel) [reply server/chat-commands/info.ts]: Hm... You could make an argument for it either way, I guess. :s

**Pull request merged**
